### PR TITLE
options page to display the banner on the site

### PR DESCRIPTION
**Pull request by robindevitt**
Hey @seagyn 

I have created an options page where the user can select a banner position as to where they would like.

I think this can be an added feature to the short-code.

Let me know any further changes or suggestions I can help with...

### CODE REVIEW

File: south-african-covid-19-banner.php [hunk 1]
```diff
@@ -48,6 +48,11 @@ function init() {
...
 	define( 'SOUTH_AFRICA_COVID_19_BANNER_URL', plugin_dir_url( __FILE__ ) );
 
+
 	boot();
 }
 add_action( 'plugins_loaded', 'SeagynDavis\SouthAfricaCovid19Banner\init' );
+
+register_activation_hook( __FILE__, 'SeagynDavis\SouthAfricaCovid19Banner\Admin\plugin_activation' );
+
+add_action( 'admin_notices', 'SeagynDavis\SouthAfricaCovid19Banner\Admin\admin_activation_notice' );
```
Comment (seagyn) [inline south-african-covid-19-banner.php]: Can you throw this into the `boot()` function, please?

File: src/admin.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * Add another activation message.
+ * Includes a URL to the settings page
+ */
+function admin_activation_notice(){
+
+  /* Check transient, if available display notice */
+  if( get_transient( 'admin_activation_notice' ) ){
+
+    // Thank you message when activat
+    echo '<div class="updated notice is-dismissible">'.
```
Comment (seagyn) [inline src/admin.php]: Can you drop this into a view (in the HTML folder), please? Should be something like `require html/...`

File: src/boot.php [hunk 3]
```diff
@@ -9,16 +9,32 @@
...
 	add_shortcode( 'south-african-covid-19-banner', 'SeagynDavis\SouthAfricaCovid19Banner\Shortcode\shortcode' );
+
 	add_action(
 		'widgets_init',
 		function() {
 			register_widget( 'SeagynDavis\SouthAfricaCovid19Banner\Widget\Banner' );
 		}
 	);
+
+	add_action( 'admin_init', 'SeagynDavis\SouthAfricaCovid19Banner\SettingsPage\rbd_settings_init' );
```
Comment (seagyn) [inline src/boot.php]: Don't worry about prefixing things because they're in a namespace.

File: src/shortcode.php [hunk 4]
```diff
@@ -13,7 +13,7 @@
  * @return string HTML to output.
  */
 function shortcode() {
-	\wp_enqueue_style( 'south-african-covid-19-banner', SOUTH_AFRICA_COVID_19_BANNER_URL . 'resources/css/south-african-covid-19-banner.css', null, SOUTH_AFRICA_COVID_19_BANNER_VERSION, false );
+	\wp_enqueue_style( 'south-african-covid-19-banner', SOUTH_AFRICA_COVID_19_BANNER_URL . 'resources/css/south-african-covid-19-banner.min.css', null, SOUTH_AFRICA_COVID_19_BANNER_VERSION, false );
```
Comment (seagyn) [inline src/shortcode.php]: We probably need to drop the minifications into a build step but don't worry about that now.

File: resources/css/south-african-covid-19-page-banner.css [hunk 5]
```diff
@@ -0,0 +1,115 @@
+body.covid_top_scroll,
```
Comment (seagyn) [inline resources/css/south-african-covid-19-page-banner.css]: So the only thing here is the WP coding standards for PHP and CSS. Check them out here:

https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/
https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/

I'll try drop in some methods to automate the testing of this.

**Review by seagyn — CHANGES_REQUESTED** [review]

File: south-african-covid-19-banner.php (hunk 1 above)
Comment (robindevitt) [reply south-african-covid-19-banner.php]: This wasn't running in the `boot()`function, it isn't critical so I have taken it out in my latest commit.

File: src/admin.php (hunk 2 above)
Comment (robindevitt) [reply src/admin.php]: I have taken this out, based on removing the activation notices.

File: resources/css/south-african-covid-19-page-banner.css (hunk 5 above)
Comment (robindevitt) [reply resources/css/south-african-covid-19-page-banner.css]: I have made some changes which I think should be fine now.

File: src/admin/sa-covid-page.php [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+// call back with the form
+function add_form( $args ) {
+
+  // get the value of the setting we've registered with register_setting()
+  $options = get_option( 'rbd_options' );
+
+  ?>
+
+  <select id="<?php echo esc_attr( $args['label_for'] ); ?>"
```
Comment (seagyn) [inline src/admin/sa-covid-page.php:68]: Can you add an option to keep it off and default to this, please?

**Review by seagyn — CHANGES_REQUESTED** [review]
Last change.

We could still improve on the CSS styling but we can come back to that lateer.

File: src/html/banner-display.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  $options = get_option( 'rbd_options' );
+
+  if ( !in_array( $options[ 'rbd_position' ], $bottom_banner_array ) ){
+    return;
+  }
+
+  banner();
+}
+
+function banner(){
```
Comment (seagyn) [inline src/html/banner-display.php]: Just picked this up now. In here, there should only be HTML. So if you can use this for HTML (like in mine) and then have these functions in src (like I have for widget/shortcode).

File: src/boot.php [hunk 8]
```diff
@@ -9,16 +9,29 @@
...
 		function() {
 			register_widget( 'SeagynDavis\SouthAfricaCovid19Banner\Widget\Banner' );
 		}
 	);
+
+	add_action( 'admin_init', 'SeagynDavis\SouthAfricaCovid19Banner\SettingsPage\settings_init' );
+
+ 	add_action( 'admin_menu', 'SeagynDavis\SouthAfricaCovid19Banner\SettingsPage\options_page' );
+
+	add_action( 'wp_head', 'SeagynDavis\SouthAfricaCovid19Banner\BannerDisplay\header_banner' );
```
Comment (seagyn) [inline src/boot.php]: I don't think this will have the desired effect because this is usually in `<head>`. Perhaps just rely on the CSS to position it?

**Review by seagyn — APPROVED** [review]

**Pull request merged**